rusqlite_migration/src/asynch.rs

```diff
@@ -29,6 +32,25 @@ impl AsyncMigrations {
         }
     }
 
+    /// Proxy implementation of the same method in the [Migrations](crate::Migrations::from_directory) struct.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rusqlite_migration::AsyncMigrations;
+    /// use include_dir::{Dir, include_dir};
+    ///
+    /// static MIGRATION_DIR: Dir = include_dir!("../examples/from-directory/migrations");
+    /// let migrations = AsyncMigrations::from_directory(&MIGRATION_DIR).unwrap();
+    /// ```
+    #[allow(clippy::missing_errors_doc)]
+    #[cfg(feature = "from-directory")]
+    pub fn from_directory(dir: &'static Dir<'static>) -> Result<Self> {
```

Comment:
I think it would be simpler for the user of the API if we just passed the path here (so taking a `&str` instead of a `Dir`). And then we would call the macro from inside the function.

Also, reading the documentation of [include_dir](https://docs.rs/include_dir/latest/include_dir/):
> When invoking the [include_dir!()](https://docs.rs/include_dir/latest/include_dir/macro.include_dir.html) macro you should try to avoid using relative paths because rustc makes no guarantees about the current directory when it is running a procedural macro.
>
> Environment variable interpolation can be used to remedy this. You might want to read the [Environment Variables](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) section of The Cargo Book for a list of variables provided by cargo.
>
> Most crates will want to use the $CARGO_MANIFEST_DIR or $OUT_DIR variables. For example, to include a folder relative to your crate you might use include_dir!("$CARGO_MANIFEST_DIR/assets").

So there are dragons here, we should probably give an example parameter like `$CARGO_MANIFEST_DIR/src/migrations` and explain that this will starts from the root of the project and descend into `src/migrations` (presumably, I’ve not actually tested. Also, unclear what happens when there is a workspace, presumably, it’s the deepest Cargo.toml that’s taken into account if I read [cargo doc](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) correctly)

Reply:
(of course this is not specific to the async version)

Reply:
I don’t know, I’ll think more about this.

Reply:
Yeah, the lifetimes were a problem here, that's why I went with this approach as it:
1. makes it clear the loading is done by `include_dir` with all it's specific behaviour details, which we don't need to document since they've done it,
2. lifetimes can be `'static`.

I do prefer the the `&str` version as well, I'll also think about it some more.

Reply:
Hmm... what about having two versions:
1. The current one, which takes `Dir`
2. Another one, let's call it `Migrations::from_migrations_directory()` or something, which just loads the `migrations` directory from the package root?

Reply:
We can always provide our own macro, which would use `include_dir` behind the scenes and do something like this:

```rust
let migrations: Migrations = from_directory!("....");
```

Reply:
Third possible option: avoid this feature altogether, provide the loader code as an example how to do it if someone wants to.

Reply:
> Third possible option: avoid this feature altogether, provide the loader code as an example how to do it if someone wants to.

I think it would be a valuable feature though.

> Yeah, the lifetimes and implementation of include_dir were a problem here, that's why I went with this approach as: […]
> 2. this makes it clear the loading is done by include_dir with all it's specific behaviour details, which we don't need to document since they've done it,

That’s a good point, it’s probably clearer and easier to debug if the user handles `include_dir` themselves, as they will have the chance to check the doc and read about rustc quirks.

Reply:
So yeah, I guess the best option is to keep what you have done, perhaps just refining the examples to include absolute path with `$CARGO_MANIFEST_DIR`.